x-pack/filebeat/tests/integration/managerV2_test.go

```diff
@@ -337,7 +338,7 @@ func TestFailedOutputReportsUnhealthy(t *testing.T) {
 	)
 
 	require.Eventually(t, func() bool {
-		return finalStateReached
+		return finalStateReached.Load()
```

Comment:
I don't quite understand why it was a race condition before. There is only one spot where we store only one value and another spot that repeatedly reads the variable. It should not have been a race condition.

Reply:
Those two spots run on different goroutines, so they can cause a race condition. One of the tests got flaky because of that: https://github.com/elastic/beats/issues/36192.

They all have the same pattern: a variable that is modified by the manager running in a goroutine and the eventually loop running in another.

Reply:
Just run `go test -race ./...` and the race detector will spot them all.